dozer-types/protos/cloud.proto

```diff
@@ -52,93 +40,103 @@ service DozerCloud {
-  // if not specified, will get latest version
-  optional uint32 version = 2;
+message DeployAppRequest {
+  optional string app_id = 1;
+  repeated File files = 2;
+  repeated Secret secrets = 3;
+  bool allow_incompatible = 4;
+  InfraRequest infra = 5;
 }
-message DeploymentStatusResponse {
-  repeated DeploymentHealth k8_deployments = 1;
-  repeated string errors = 2;
-  map<uint32, uint32> versions = 3;
-  uint32 current_version = 4;
+
+message InfraRequest {
+  int32 num_api_instances = 2;
 }
 
-message DeploymentHealth {
-  string name = 1;
-  string status = 2;
-  string created_at = 3;
-  int32 desired = 4;
-  int32 available = 5;
+message DeployStep {
+  bool is_completed = 1;
+  string text = 2;
+  uint32 current_step = 3;
+  optional string error = 4;
 }
 
-message AppConnectionRequest {
-  string app_id = 1;
-  oneof input {
-    Connection connection = 2;
-    string connection_yaml = 3;
+message DeployAppResponse {
+  oneof result {
+    AppResponse app_created = 1;
+    DeployStep step = 2;
+    string log_message = 4;
   }
+  bool completed = 5;
+  optional string error = 6;
 }
 
-message CreateSourceRequest {
+message AppResponse {
   string app_id = 1;
-  Source source = 2;
-  Connection connection = 3;
+  AppConfig app = 2;
+  optional string repository_url = 3;
+  repeated DeploymentInfo deployments = 4;
+  google.protobuf.Timestamp created_at = 5;
+  google.protobuf.Timestamp updated_at = 6;
 }
 
-message ListEndpointRequest {
-  string app_id = 1;
-  optional uint32 limit = 2;
-  optional uint32 offset = 3;
+message ListAppRequest {
+  optional uint32 limit = 1;
+  optional uint32 offset = 2;
+  optional string name = 3;
+  optional string uuid = 4;
+  optional string order_by = 5;
+  optional bool desc = 6;
 }
-
-message ListEndpointResponse {
-    repeated EndpointResponse endpoints = 1;
-    Pagination pagination = 2;
+message ListAppResponse {
+  repeated AppResponse apps = 1;
+  Pagination pagination = 2;
 }
 
-message EndpointResponse {
-  string id = 1;
-  Endpoint endpoint = 2;
-  string yaml_content = 3; 
+message UpdateAppRequest {
+  string app_id = 1;
+  repeated File files = 2;
 }
 
-message ListSourceRequest {
+message DeleteAppRequest {
   string app_id = 1;
-  optional uint32 limit = 2;
-  optional uint32 offset = 3;
 }
+message DeleteAppResponse { bool success = 1; }
 
-message ListSourceResponse {
-    repeated SourceResponse sources = 1;
-    Pagination pagination = 2;
-}
+message GetAppRequest { optional string app_id = 1; }
 
-message SourceResponse {
-  string id = 1;
-  Source source = 2;
-  string yaml_content = 3;
+message AppHealthRequest {
+  string app_id = 1;
 }
-
-message ListAppConnectionRequest {
+message AppHealthResponse {
+  bool healthy = 1;
+}
+message GetResourcesRequest {
   string app_id = 1;
-  optional uint32 limit = 2;
-  optional uint32 offset = 3;
+  // if not specified, will get latest version
+  optional uint32 version = 2;
+}
+message ResourcesResponse {
```

Comment:
Looks like renamed from `DeploymentStatusResponse`. Why remove `DeploymentStatusResponse::errors`?

Reply:
I get the state reduction part. It's just that we're not getting errors anymore.

Reply:
These errors were not really errors. They were just returning the last message in each pod which was rather confusing. 
Instead we should have an API that either calls`get_logs` or `get_errors` and create a separate view for it.

Reply:
Makes sense.